### PR TITLE
Add freetype to fastqc deps

### LINKED ISSUE #5026: Missing ttf-dejavu/fontconfig dependencies for fastqc
If fontconfig or ttf-dejavu are not installed, fastqc can produce the following error:

```
java.lang.Error: Probable fatal error:No fonts found
```

which is identical to the error described here: https://stackoverflow.com/questions/14066525/java-lang-error-probable-fatal-errorno-fonts-found. This issue is at least encountered in the snakemake container used by snakemake-wrappers.

I think this error is not necessarily specific to fastqc, but likely a more general dependency of openjdk which is currently missing. 

We can easily fix it for fontconfig by adding the dependency to the fastqc or openjdk recipes. However, for ttf-dejavu there does not seem to be an existing package as far as I can see. I have no experience with creating non-R/Python packages, so was wondering if someone would be prepared to build a ttf-dejavu package or otherwise provide me with pointers on how to do so.

### PR DESCRIPTION
**Pull request by epruesse**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Should fix #5026

### CODE REVIEW

File: recipes/fastqc/meta.yaml [hunk 1]
```diff
@@ -6,13 +6,14 @@ about:
...
   detect_binary_files_with_prefix: true
-  number: 2
+  number: 3
   skip: False
 
 requirements:
   run:
     - openjdk
     - perl
+    - freetype 2.7
```
Comment (bgruening) [inline recipes/fastqc/meta.yaml]: This is strange. fastqc is just a Java program and should not depend on it. Is it missing for openjdk or perl?

**Comment by epruesse** [comment]
The conda-forge openjdk package is "missing" a number of library dependencies for "full" functionality. These are the freetype lib needed for font rendering as done by fastqc and a bunch of X11 libs for screen rendering. Since the JVM works fine without those, it's a matter of taste whether they need to be included by packages using Java or Java itself, potentially though meta-packages (say openjdk-x11, openjdk-fonts). To get fastqc working asap I'm just adding it here. 

I did post an issue at conda forge a while ago ... see https://github.com/conda-forge/openjdk-feedstock/issues/19

**Comment by epruesse** [comment]
@bgruening Can I merge this? It should fix the issue in the container.

**Comment by bgruening** [comment]
Go for it. I'm still not a fan of this and think this should go upstream to open.jdk - but its important to fix it now.

**Comment by mbargull** [comment]
How about adding a short explanatory comment to that dependency?

**Comment by epruesse** [comment]
Hmm. This is weird. Technically, the package shouldn't need freetype. conda-forge::openjdk uses the azul/zulu java packages which bundle the required libs including freetype. The linking seems to be ok too. So in theory it should work without.

That said, I did have the problem in my pipeline in the unit tests and adding freetype fixed it there...

**Comment by epruesse** [comment]
Ok, looks like the quay container is using the defaults channel java, rather than the conda-forge one. 
Closing this for now.

**Pull request closed without merging**